pkg/broker/broker.go

```diff
@@ -261,16 +305,57 @@ func (a AnsibleBroker) Bootstrap() (*BootstrapResponse, error) {
 				log.Errorf("We have a plan that did not get its id generated: %v", p.Name)
 				continue
 			}
-			planNameManifest[p.ID] = p.Name
 		}
 	}
-	if err := a.dao.BatchSetSpecs(specManifest); err != nil {
-		return nil, err
+	return specManifest, updateSpecManifest
+}
+
+// getSafeToDeleteSpecs - will check if any bundle instance spe
+func getSafeToDeleteSpecs(a AnsibleBroker, markedSpecs map[string]*bundle.Spec) []*bundle.Spec {
+	safeToDeleteSpecs := make([]*bundle.Spec, 0)
+	bundleInstances, err := a.dao.BatchGetBundleInstances()
+	if err != nil {
+		log.Errorf("error getting bundle instances '%+v'", err)
+		// returning nil because the instead of checking the error,
+		// the broker can simply ignore to delete the specs and hope
+		// that the error wont repeat during next bootstrap cycle
+		return nil
+	}
+	log.Debugf("markedSpecs: %+v\n", markedSpecs)
+	for _, bundleInstance := range bundleInstances {
```

Comment:
We could be going through this loop even after all the specs have been removed from the map. 

Could we add a check at the bottom that could then break out of the loop?

Reply:
Added a condition to break out if the markedSpec map is empty.